[Glib] Initialize `WebProcessExtension`s in API tests on demand

obyknovenius opened the pull request:
#### f7d2305bf1ed1a256d04dfd67db540cb28ba221f
<pre>
[Glib] Initialize `WebProcessExtension`s in API tests on demand
<a href="https://bugs.webkit.org/show_bug.cgi?id=260382">https://bugs.webkit.org/show_bug.cgi?id=260382</a>

Reviewed by Carlos Garcia Campos and Michael Catanzaro.

Currently, `WebProcessExtension`s are initialized for each API test.
It&apos;s not only unnecessary but can lead to flakiness.

This patch requires API tests using extensions to explicitly state this
by setting `Test::shouldInitializeWebProcessExtensions` to `true`.

* Tools/TestWebKitAPI/Tests/WebKitGLib/TestConsoleMessage.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestDOMElement.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestEditor.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestFrame.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestLoaderClient.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestMultiprocess.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestResources.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitUserContentManager.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebProcessExtensions.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestAutocleanups.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestContextMenu.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMClientRect.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMNode.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMNodeFilter.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMXPathNSResolver.cpp:
(beforeAll):
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.cpp:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::Test):

Canonical link: <a href="https://commits.webkit.org/267125@main">https://commits.webkit.org/267125@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30eaf31f2e299c952573d13262872552df6a47a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15655 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15961 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16334 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17411 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14690 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15837 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18925 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16063 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17213 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15840 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16313 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13326 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18156 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13574 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14128 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21049 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14577 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14294 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17555 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14886 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12620 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14142 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3759 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18504 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14706 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

TingPing commented on Tools/TestWebKitAPI/Tests/WebKitGLib/TestMultiprocess.cpp:
```diff
@@ -39,6 +49,7 @@ class MultiprocessTest: public Test {
 
     void initializeWebProcessExtensions() override
     {
+        g_print("initializeWebProcessExtensions()\n");
```
Debug print or intentional?

obyknovenius replied:
Oops. No, it wasn't intentional.

obyknovenius replied:
Removed it.

obyknovenius committed:
[Glib] Initialize `WebProcessExtension`s in API tests on demand
https://bugs.webkit.org/show_bug.cgi?id=260382

Reviewed by Carlos Garcia Campos and Michael Catanzaro.

Currently, `WebProcessExtension`s are initialized for each API test.
It's not only unnecessary but can lead to flakiness.

This patch requires API tests using extensions to explicitly state this
by setting `Test::shouldInitializeWebProcessExtensions` to `true`.

* Tools/TestWebKitAPI/Tests/WebKitGLib/TestConsoleMessage.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestDOMElement.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestEditor.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestFrame.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestLoaderClient.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestMultiprocess.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestResources.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitUserContentManager.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebProcessExtensions.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestAutocleanups.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestContextMenu.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMClientRect.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMNode.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMNodeFilter.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMXPathNSResolver.cpp:
(beforeAll):
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.cpp:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::Test):

Canonical link: https://commits.webkit.org/267125@main
